src/plugins/flexibleLayout/plugin.js

```diff
@@ -0,0 +1,118 @@
+                creatable: true,
+                description: "A fluid, flexible layout canvas that can display multiple objects in rows or columns.",
+                cssClass: 'icon-flexible-layout',
+                initialize: function (domainObject) {
+                    domainObject.configuration = {
+                        containers: [],
+                        rowsLayout: false
+                    };
+                }
+            });
+
+            openmct.toolbars.addProvider({
+                name: "Flexible Layout Toolbar",
+                key: "flex-layout",
+                description: "A toolbar for objects inside a Flexible Layout.",
+                forSelection: function (selection) {
+                    let parent = selection[selection.length - 1];
+
+                    return (parent && parent.context.item &&
+                        parent.context.item.type === 'flexible-layout' &&
+                        openmct.editor.isEditing());
+                },
+                toolbar: function (selection) {
+                    let domainObject = selection[selection.length - 1].context.item;
+
+                    let deleteButton = {
+                            control: "button",
+                            domainObject: selection[0].context.item,
+                            method: selection[0].context.method,
+                            key: "remove",
+                            icon: "icon-trash",
+                            title: `Remove ${selection[0].context.type}`
+                        },
+                        toggleButton = {
+                            control: 'toggle-button',
+                            key: 'toggle-layout',
+                            domainObject: domainObject,
+                            property: 'configuration.rowsLayout',
+                            options: [
+                                {
+                                    value: false,
+                                    icon: 'icon-columns',
+                                    title: 'Columns'
+                                },
+                                {
+                                    value: true,
+                                    icon: 'icon-rows',
+                                    title: 'Rows'
+                                }
+                            ]
+                        },
+                        addContainerButton = {
+                            control: "button",
+                            domainObject: selection[0].context.item,
+                            method: selection[0].context.addContainer,
+                            key: "add",
+                            icon: "icon-plus-in-rect",
+                            title: 'Add Container'
+                        },
+                        separator = {
+                            control: "separator",
+                            domainObject: selection[0].context.item,
```

Comment:
I don't believe domain object or key is necessary for separator.

Reply:
It does, throws an error if domainObject is not present